src/lambdaconnect_model/scoping.clj

```diff
@@ -328,29 +310,125 @@
 ;                              INTERFACE
 ; ========================================================================================
 
+(defn scope-selected-tags-with-tree
+  "Takes a snapshot, a user object from DB, entities-by-name, parsed (and validated) EDN of scoping rules, map of sets indiacting which tags must be scoped per tag and a set of desired tags.
+   It is advised to calculacte scoping sets once and pass the result.
+  A typical invocation looks like this: 
+  (scope-selected-tags-with-tree config (d/db db/conn) user entities-by-name validated-scope scoping-sets #{:RARestaurant.ofOwner})))
+
+  Returns a map with db ids, something like:
+  {:RAOwner.me #{11122, 1222} :user #{2312312}}
+  "
+  [config snapshot user entities-by-name scoping-defintion tag-scope tags]
+  (let [relevant-rules (->> tags
+                            (map (fn [tag]
+                                   (->> scoping-defintion
+                                        (filter (fn [[k _]] (contains? (tag tag-scope) k)))
+                                        (map (fn [[tag description]] [tag (:constraint description)]))
+                                        (into {}))))
+                            (reduce merge))
+        queries (scoping-step
+                 snapshot
+                 #{(:db/id user)}
+                 entities-by-name
+                 {}
+                 #{:user}
+                 relevant-rules
+                 {:user {:dependencies #{} :rules []}})
+        filtered-queries (into {} (filter (fn [[tag _]] (contains? tags tag)) queries))]
+    (->> filtered-queries
+         (pmap (fn [x] (apply (partial execute-query config) x)))
+         (into {}))))
+
+(defn scope-selected-tags
+  "Takes a snapshot, a user object from DB, entities-by-name, parsed (and validated) EDN of scoping rules, a set of desired tags and push?.
```

Comment:
zacząłem się zastanawiać nad tym parametrem `push?` i `include-in-push` w scopingu -- być może za jakiś czas trzeba będzie się tego pozbyć

Reply:
może nie do końca wiem po co on jest i czy jest ważny więc ni tykam

Reply:
jasne, ja się postaram to zmienić